src/components/ContentTemplate.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+	<NcAppContent :show-details="showDetails"
+		@update:showDetails="unselectItem()">
+		<template #list>
+			<NcAppContentList>
+				<div class="header">
+					<slot name="header" />
+				</div>
+
+				<FeedItemDisplayList :items="items"
+					:fetch-key="fetchKey"
+					:config="config"
+					@load-more="emit('load-more')" />
+			</NcAppContentList>
+		</template>
+
+		<NcAppContentDetails class="content-details-container">
+			<FeedItemDisplay v-if="selectedFeedItem" :item="selectedFeedItem" />
+		</NcAppContentDetails>
+	</NcAppContent>
+</template>
+
+<script setup lang="ts">
+
+import { PropType, computed, ref, watch } from 'vue'
+
+import itemStore from '../store/item'
+
+import NcAppContent from '@nextcloud/vue/dist/Components/NcAppContent'
+import NcAppContentList from '@nextcloud/vue/dist/Components/NcAppContentList'
+import NcAppContentDetails from '@nextcloud/vue/dist/Components/NcAppContentDetails'
+
+import { FeedItem } from '../types/FeedItem'
+
+import FeedItemDisplayList from './feed-display/FeedItemDisplayList.vue'
+import FeedItemDisplay from './feed-display/FeedItemDisplay.vue'
+
+defineProps({
+	items: {
+		type: Array as PropType<Array<FeedItem>>,
+		required: true,
+	},
+	fetchKey: {
+		type: String,
+		required: true,
+	},
+	config: {
+		type: Object,
+	},
+})
+
+const emit = defineEmits<{
+    (event: 'load-more'): void
+  }>()
+
+const showDetails = ref(false)
+
+const selectedFeedItem = computed(() => {
+	return itemStore.getters.selected(itemStore.state)
```

Comment:
is this best practice to access the itemStore via import rather than with the `$store` property on the component?

Reply:
I don't know, but since you used v3 of `vuex` and I didn't want to upgrade there is no `useStore` function available. And I couldn't figure out another way to access `$store` with composition API..

Reply:
Do you think it would be a hassle to update vuex? 

I think it would be best to use the composition api with as best practices as possible in this file so I can reference it as the preferred way to create components.

if it will be too much though, I can try to update it and this in a follow up PR..

Reply:
I can have a look at it, but I did not do it all in this PR since it's not directly related. We could leave this now, and I change it later with the next PR..

Reply:
It seems that Vuex v4 is for Vue 3 only, so we are stuck with v3 until Nextcloud has moved as well..

Reply:
ah.. I do sort of remember running into that when I was working on the rewrite. Makes sense why we can't do it then.

Reply:
Also I found that `pinia` (https://pinia.vuejs.org/) is supposed to be the successor to `vuex` and it should work with vue 2.7 too. Also looks a lot simpler, maybe we can convert to that..? In a separate PR of course..